src/Twig/TwigFileNormalizer.php

```diff
@@ -16,19 +16,24 @@ public function __construct(
 
     public function normalize(string $twigFileName): string
     {
+        if (str_starts_with($twigFileName, '$')) {
+            return $twigFileName;
+        }
+
         if (str_starts_with($twigFileName, '@')) {
             return $twigFileName;
         }
 
-        if (Path::isAbsolute($twigFileName)) {
-            /**
-             * @var AbsolutePathLoader $loader
-             */
-            $loader = $this->twig->getLoader();
+        /**
+         * @var AbsolutePathLoader $loader
+         */
+        $loader = $this->twig->getLoader();
 
-            return $loader->maybeResolveAbsolutePath($twigFileName);
-        }
+        return $loader->maybeResolveAbsolutePath($twigFileName);
+    }
 
-        return sprintf('@%s', $twigFileName);
+    public function toAbsolute(string $twigFileName): string
+    {
+        return $this->twig->getLoader()->getSourceContext($twigFileName)->getPath();
```

Comment:
The annoying thing is that this does load the template into memory, only to get the path. 
https://github.com/twigphp/Twig/pull/4276